src/config_mem.c

```diff
@@ -198,7 +198,7 @@ int git_config_backend_from_string(git_config_backend **out, const char *cfg)
 		return -1;
 	}
 
-	if (git_buf_sets(&backend->cfg, cfg) < 0) {
+	if (git_buf_put(&backend->cfg, cfg, len) < 0) {
```

Comment:
This shouldn't be a put but a `git_buf_set` instead. Put would append the string, while set will replace the current string.